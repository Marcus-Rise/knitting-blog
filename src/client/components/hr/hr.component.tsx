import React from "react";
import styled from "styled-components";

const Hr = styled.hr`
  height: 0;
  border-top: 0.1rem dashed;
  max-width: 80%;
`;

export { Hr };
