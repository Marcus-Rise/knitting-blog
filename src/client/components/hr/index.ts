export { Hr } from "./hr.component";
