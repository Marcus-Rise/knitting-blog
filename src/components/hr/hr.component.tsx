import React from "react";
import styles from "./hr.module.scss";

const Hr: React.FC = () => <hr className={styles.hr} />;

export { Hr };
