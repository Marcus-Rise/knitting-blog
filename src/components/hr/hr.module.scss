.hr {
  border: 1px dashed;
  max-width: 80%;
}
