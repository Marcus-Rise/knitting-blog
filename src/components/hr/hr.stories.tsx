import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { Hr } from "./hr.component";

const Config: ComponentMeta<typeof Hr> = {
  title: "components/Hr",
  component: Hr,
};

const Template: ComponentStory<typeof Hr> = (args) => <Hr {...args} />;

const Default = Template.bind({});

export default Config;
export { Default };
