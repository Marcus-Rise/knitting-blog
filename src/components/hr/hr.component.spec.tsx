import { render } from "@testing-library/react";
import { Hr } from "./hr.component";
import React from "react";

describe("Hr", () => {
  test("render", () => {
    const { asFragment } = render(<Hr />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("styles", () => {
    const { asFragment } = render(<Hr styles={{ marginTop: "3rem", marginBottom: "3rem" }} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
