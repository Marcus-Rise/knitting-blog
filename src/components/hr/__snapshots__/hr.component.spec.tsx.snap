// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hr render 1`] = `
<DocumentFragment>
  <hr
    class="hr"
  />
</DocumentFragment>
`;
